### PR TITLE
Detecting loops in Sankeys

### PR DESCRIPTION
**Pull request by mistercrunch**
screen shot 2016-04-05 at 8 48 32 am
@williaster

### CODE REVIEW

File: caravel/viz.py [hunk 1]
```diff
@@ -994,7 +994,20 @@ def query_obj(self):
     def get_data(self):
         df = self.get_df()
         df.columns = ['source', 'target', 'value']
-        return df.to_dict(orient='records')
+        recs = df.to_dict(orient='records')
+        seen = set()
+        hierarchy = {row['source']: row['target'] for row in recs}
+        def find_loop(source, target):
```
Comment (williaster) [inline caravel/viz.py]: might just be me but comments might make this a little more readable. not huge deal.

**Comment by williaster** [comment]
super nice add / nice debugging! 

LGTM overall, though I can't tell if the failed tests were intentional or not. Detecting sankey loops seems like a good opportunity for a test(s).

**Comment by mistercrunch** [comment]
Yeah I'm confused, apparently there's a loop in the example, but how does it still work then! I'll have to dig deeper

**Comment by williaster** [comment]
see the slack channel for another example of a broken sankey, cc-ed you.

**Pull request merged**